Buildings/Fluid/BuriedPipes/GroundCoupling.mo

```diff
@@ -0,0 +1,270 @@
+          lineThickness=0.5,
+          fillColor={122,20,25},
+          fillPattern=FillPattern.Solid,
+          textString="pos[3]"),
+        Text(
+          extent={{7,-3},{-7,3}},
+          lineColor={0,0,0},
+          lineThickness=0.5,
+          fillColor={122,20,25},
+          fillPattern=FillPattern.Solid,
+          textString="dep[1]",
+          origin={-85,7},
+          rotation=90),
+        Text(
+          extent={{7,-3},{-7,3}},
+          lineColor={0,0,0},
+          lineThickness=0.5,
+          fillColor={122,20,25},
+          fillPattern=FillPattern.Solid,
+          origin={-25,7},
+          rotation=90,
+          textString="dep[2]"),
+        Text(
+          extent={{7,-3},{-7,3}},
+          lineColor={0,0,0},
+          lineThickness=0.5,
+          fillColor={122,20,25},
+          fillPattern=FillPattern.Solid,
+          origin={35,7},
+          rotation=90,
+          textString="dep[3]"),
+        Text(
+          extent={{-64,-2},{-52,-8}},
+          lineColor={0,0,0},
+          lineThickness=0.5,
+          fillColor={122,20,25},
+          fillPattern=FillPattern.Solid,
+          textString="rad[1]"),
+        Text(
+          extent={{0,-32},{12,-38}},
+          lineColor={0,0,0},
+          lineThickness=0.5,
+          fillColor={122,20,25},
+          fillPattern=FillPattern.Solid,
+          textString="rad[2]"),
+        Text(
+          extent={{54,-14},{66,-20}},
+          lineColor={0,0,0},
+          lineThickness=0.5,
+          fillColor={122,20,25},
+          fillPattern=FillPattern.Solid,
+          textString="rad[3]"),
+        Line(
+          points={{-84,-52},{-84,-92}},
+          color={0,0,0},
+          thickness=0.5)}), Documentation(info="<html>
+<p>
+This model simulates the heat transfer between multiple buried pipes and the ground,
+using climate and soil information and the geometry of the pipes network.
+
+The heat transfer solution is based upon the potential flow theory and obtained by the use
+of \"mirror-image\" technique suggested by Eckert (1959). This technique is extended to
+a network with multiple pipes by Kusuda (1981) in the equation:
+</p>
+<p align=\"center\">
+<img alt=\"image\" src=\"modelica://Buildings/Resources/Images/Fluid/BuriedPipes/BuriedPipesHeatTransfer.png\" />
+</p>
+<p>
+where k<sub>s</sub> is the soil thermal conductivity, P<sub>ij</sub> is a 
+geometric factor that is function of the pipes and mirror images positions (see 
+<code>Buildings.Fluid.BuriedPipes.Functions.groundCouplingFactors</code> for more information),
+Q<sub>i</sub> is the heat transfer from the ith pipe, T<sub>i</sub> is the temperature at
+the exterior surface of the ith pipe, and T<sub>g</sub> is the undisturbed ground 
+temperature at the depth of the network.
+</p>
+<p>
+This model relies on the following assumptions.
+</p>
+<ul>
+<li>Heat transfer is in steady state (although seasonal heat storage
+is modeled in the ground temperature model).</li>
```

Comment:
I am not quite sure how to answer your initial question. The initial simplification that goes into calculating the undisturbed ground temperature is that the surface temperature is an annual sinusoid. This simplification is going to be more or less accurate at smaller time scale depending on your climate, but typically would fail to precisely encompass daily variation.
As you go deeper into the ground, the daily temperature variation are less important, and the model would be more accurate on smaller time scale. For instance, depending on the soil properties, if you are 10m deep, you might expect the ground to be almost at a constant temperature year round, and the model would accurately represent that at even the second time-scale.

![image](https://user-images.githubusercontent.com/48033702/124680164-acb74380-de7a-11eb-9a62-ae50993af74b.png)

With that in mind, the statement about the heat transfer being steady state is actually meant to qualify the heat transfer between the pipes and the soil. The temporal scale at which this is relevant depends on the problem. If the pipe temperature vary significantly throughout the day and the user need to project hourly data, this might not be an accurate representation. On the other hand, if the fluid temperature is relatively stable, and the pipe is either well insulated or deep enough in the ground (so that the undisturbed ground temperature model is accurate at faster time scale), then the model might accurately represent steady-state heat losses.

What I meant by "seasonal heat storage is modeled in the ground temperature model" is probably poorly formulated, but is meant to convey that the undisturbed ground temperature is varying throughout the year, and that its calculation takes into account the thermal diffusivity of the soil.